great_expectations/experimental/datasources/interfaces.py

```diff
@@ -371,94 +380,88 @@ def execution_engine_type(self) -> Type[ExecutionEngine]:
     # End Abstract Methods
 
 
-class Batch:
-    # Instance variable declarations
-    _datasource: Datasource
-    _data_asset: DataAsset
-    _batch_request: BatchRequest
-    _data: BatchDataType
-    _id: str
+class Batch(ExperimentalBaseModel):
+    """This represents a batch of data.
+
+    This is usually not the data itself but a hook to the data on an external datastore such as
+    a spark or a sql database. An exception exists for pandas or any in-memory datastore.
+    """
+
+    datasource: Datasource
+    data_asset: DataAsset
+    batch_request: BatchRequest
+    data: "BatchData"
```

Comment:
Could we type annotate this without quotes since we use `__future__`?

Reply:
No because of circular imports. The quotes avoid the circular import issue and then `pydantic` still wants to validate the actual type, so you have to pass the type to `Batch.update_forward_refs()` in the scope that you are instantiating the `Batch` class (see changes to `pandas_datasource.py` and `sql_datasource.py`.

Reply:
I should also say using `if TYPE_CHECKING` won't help because `pydantic` actually validates at runtime (as opposed to `mypy`).